### PR TITLE
Merge protocol designer

### PR DESCRIPTION
**Pull request by IanLondon**
## overview

This PR merges the protocol designer from its own repo into the `opentrons` monorepo.

`.travis.yml` is updated to build the protocol designer app and deploy it to its own S3 bucket, under a subdirectory named after the branch.

`.gitignore` - remove `.*` line

Protocol designer now uses the same node version, '6.5.11', as the Opentrons app.

The git history of the protocol designer repo was merged in. All commits from that repo were prefixed with "protocol designer: " to make them easier to distinguish.

Additional documentation

## review requests

* How should Makefiles and travis work in the monorepo? When is the time to use make instead of the `cd protocol-designer; npm install && npm run test && npm run lint && npm run build; cd ..` line in travis?

* The protocol designer is in need of more test coverage and some refactoring to make components more readable, but I'm thinking those issues will be for a later PR. If anything jumps out at you, please address in a comment and I'll work on it for another PR.

### CODE REVIEW

File: protocol-designer/src/components/IngredientPropertiesForm.css [hunk 1]
```diff
@@ -0,0 +1,130 @@
+.ingredientPropertiesEntry {
+  background-color: white;
+  margin: 1em;
```
Comment (mcous) [inline protocol-designer/src/components/IngredientPropertiesForm.css]: We're tending to stick with `rem` over `em`s in the app

File: protocol-designer/src/components/IngredientsList.css [hunk 2]
```diff
@@ -0,0 +1,269 @@
+input {
```
Comment (mcous) [inline protocol-designer/src/components/IngredientsList.css]: I discovered the other day that using tags in CSS modules doesn't actually get scoped? I don't know if this was a misconfiguration or something but heads up

File: protocol-designer/package.json [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    "name": "Opentrons Labworks",
+    "email": "engineering@opentrons.com"
+  },
+  "name": "protocol-designer-prototype",
+  "version": "1.0.0",
+  "description": "Protocol designer prototype",
+  "main": "index.js",
+  "scripts": {
+    "test": "jest",
+    "lint": "standard",
```
Comment (mcous) [inline protocol-designer/package.json]: Noticed some minor formatting issues in some CSS files (e.g. mixed tabs and spaces). We might want to consider installing + running `stylelint` + `stylelint-config-css-modules` + `stylelint-config-standard` and giving it a run before merge.

Apps stylelint config in package.json for reference:

```json
  "stylelint": {
    "extends": [
      "stylelint-config-standard",
      "stylelint-config-css-modules"
    ],
    "ignoreFiles": [
      "**/dist/**",
      "**/coverage/**"
    ]
  },
```

File: protocol-designer/src/configureStore.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  const store = createStore(
+    reducer,
+    /* preloadedState, */
+    composeEnhancers(applyMiddleware(thunk))
+  )
+
+  if (module.hot) {
+    // Enable Webpack hot module replacement for reducers
+    module.hot.accept('./reducers/index', () => {
+      const nextRootReducer = combineReducers(require('./reducers/index'))
```
Comment (mcous) [inline protocol-designer/src/configureStore.js]: With webpack native modules (i.e. `modules: false` in babelrc) I think this require call might not be necessary (as opposed to one import of the reducer up top). I know component hot reloading in the app works without an extra require call now. Could be different for non-react modules, though, but I can't think of why that would be the case

File: protocol-designer/src/containers/LabwareContainer.js [hunk 5]
```diff
@@ -0,0 +1,177 @@
+// NOTE: This mixed container + component does a lot right now:
```
Comment (mcous) [inline protocol-designer/src/containers/LabwareContainer.js]: 😬

File: protocol-designer/src/reducers/index.js [hunk 6]
```diff
@@ -0,0 +1,457 @@
...
+import range from 'lodash/range'
+import reduce from 'lodash/reduce'
+import set from 'lodash/set' // <- careful, this mutates the object
+
+import { containerDims, toWellName, getMaxVolumes, wellKeyToXYList } from '../constants.js'
+import { uuid } from '../utils.js'
+
+const sortedSlotnames = [].concat.apply( // flatten
+  [],
+  [1, 2, 3].map(num => ['A', 'B', 'C', 'D', 'E'].map(letter => letter + num))
```
Comment (mcous) [inline protocol-designer/src/reducers/index.js]: I think we're probably better off in terms of readability to just write out the array

File: protocol-designer/webpack.config.js [hunk 7]
```diff
@@ -0,0 +1,64 @@
+var webpack = require('webpack')
```
Comment (mcous) [inline protocol-designer/webpack.config.js]: We probably shouldn't use `var`s

File: .gitignore [hunk 8]
```diff
@@ -7,6 +7,7 @@
 !api/.python-version
 !app/.babelrc
 !app/.gitignore
+!protocol-designer/.babelrc
```
Comment (mcous) [inline .gitignore]: Comment for @btmorr: this dotfile ignore is still wrecking havoc on my editor (admittedly more of an Atom problem than a config problem, but one of these things I have some immediate control over). Could we maybe consider moving back to the old way of doing things (manual blacklist of dotfiles to ignore)?

Comment (IanLondon) [reply .gitignore]: agree!

File: protocol-designer/src/reducers/index.js (hunk 6 above)
Comment (IanLondon) [reply protocol-designer/src/reducers/index.js]: haha why did I do that

File: .gitignore (hunk 8 above)
Comment (btmorr) [reply .gitignore]: I thought the switch back was already merged in?

**Pull request merged**